### PR TITLE
Readme - On Spack

### PR DESCRIPTION
**Pull request by fluidnumerics-joe**
I've added some documentation for installing FTObjectLibrary via spack. Additionally, there are notes on how to contribute new version releases for FTOL to the Spack package manager.

Ideally, you should be able to install Spack using the instructions provided.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,6 +91,51 @@ make
 
 That will create the necessary files in that directory, which can be moved to somewhere else as desired.
 
+### Updating Spack Packages
+When new releases of FTObjectLibrary are made, you can contribute the new release to the spack packages.
+Before making an update to the Spack package, you should familiarize yourself with [the Spack packaging guide](https://spack.readthedocs.io/en/latest/packaging_guide.html)
+
+
+To add a new FTObjectLibrary release to Spack,
+1. Fork the Spack repository on Github to your personal account or the trixi-framework organization account.
```
Comment (sloede) [inline README.md:100]:
```suggestion
1. Fork the Spack repository on GitHub.
```
Forks should never go into the organization account.

File: README.md [hunk 2]
```diff
@@ -77,6 +91,51 @@ make
...
+7. Test to make sure the installation works
+```
+spack install ftobjectlibrary@v1.0.1
+```
+
+8. Run the `spack style` command to ensure that you are meeting Spack's style requirements.
+
+9. When ready, commit your changes and push them to your Github repository.
+
+10. Open a Pull Request with `github.com/spack/spack` to merge your changes with the `spack/spack/develop` branch.
```
Comment (sloede) [inline README.md:137]:
```suggestion
10. Open a Pull Request against `github.com/spack/spack` to merge your changes with the `spack/spack/develop` branch.
```
I think "open a pull request *against* a repo/branch" is the typical GH lingo

**Review by sloede — CHANGES_REQUESTED** [review]
Just two minor suggestions, otherwise this LGTM!

**Review by DavidAKopriva — APPROVED** [review]
OK. Sorry I didn't know about the review request.

**Review by andrewwinters5000 — APPROVED** [review]

**Pull request merged**